internal/stacksapi/get_job.go

```diff
@@ -0,0 +1,34 @@
+package stacksapi
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+)
+
+type GetJobRequest struct {
+	StackKey string `json:"-"` // The key to call the stack. Required.
+	JobUUID  string `json:"-"` // The uuid of a job. Required.
+}
+
+type GetJobResponse struct {
+	ID      string            `json:"id"`
+	Env     map[string]string `json:"env"`
+	Command string            `json:"command"`
+}
+
+// GetJob return Job's runtime data (env and command) for a single job.
+func (c *Client) GetJob(ctx context.Context, getJobReq GetJobRequest, opts ...RequestOption) (*GetJobResponse, http.Header, error) {
+	path := fmt.Sprintf("/stacks/%s/jobs/%s", getJobReq.StackKey, getJobReq.JobUUID)
```

Comment:
Similar comment that I made in #716: if `StackKey` is customer-supplied, then we should be sure to %-escape `.` - see https://github.com/buildkite/agent/blob/4e9dc55fb68e89a0ee9c8994f3afe48d27b7d01d/api/client.go#L418-L421